### ISSUE TITLE
Waiting for SSH service

### ISSUE THREAD

**Issue by jgrill** [test-kitchen/kitchen-openstack#91]
CentOS 6.6; ChefDK > 0.4.0; kitchen-openstack 1.8.1

We've tried every version of ChefDK greater than 0.4.0 (including the latest 0.7.0) with kitchen-openstack 1.8.1 and experience a failure to detect SSH service on remote VM.

```
(server ready)
       Attaching floating IP from <net04_ext> pool
       Attaching floating IP <172.26.4.188>
       Waiting for SSH service on 172.26.4.188:22, retrying in 3 seconds
       Waiting for SSH service on 172.26.4.188:22, retrying in 3 seconds
       Waiting for SSH service on 172.26.4.188:22, retrying in 3 seconds
       Waiting for SSH service on 172.26.4.188:22, retrying in 3 seconds
      [truncated for brevity]
       Waiting for SSH service on 172.26.4.188:22, retrying in 3 seconds
       Waiting for SSH service on 172.26.4.188:22, retrying in 3 seconds
       Waiting for SSH service on 172.26.4.188:22, retrying in 3 seconds
$$$$$$ [SSH] connection failed, terminating (#<Net::SSH::AuthenticationFailed: Authentication failed for user cloud-user@172.26.4.188>)
>>>>>> ------Exception-------
>>>>>> Class: Kitchen::ActionFailed
>>>>>> Message: Failed to complete #create action: [SSH session could not be established]
```

Downgrading to ChefDK 0.4.0 and rerunning the same operation works just fine. Not sure if it helps to know this, but I can edit the .kitchen/default-centos-66.yml and modify the last action and ssh key and `kitchen login` will work, which leads me to believe it's just a problem with initially detecting the ssh service.

**Comment by jjasghar** [comment]
I was planning on dedicating a huge chunk of time next week to kitchen-OpenStack. I'll keep this issue in mind as I attack it.

**Comment by jgrill** [comment]
Awesome!  Thank you.

**Comment by jjasghar** [comment]
@jgrill can you humor me and try out https://github.com/test-kitchen/kitchen-openstack/issues/93 and see if this fixes your problem?

**Comment by jgrill** [comment]
Sorry for the delay. I'll try it out as soon as I can and let you know.  Tnx!

**Comment by jgrill** [comment]
Still having the same problem.  I see you created a new 2.x dev release.  Will try that and let you know.

**Comment by jgrill** [comment]
No dice.  Were you able to duplicate?

**Comment by jjasghar** [comment]
Yeah try the 2.x release, with the new transport this should fix this problem.

**Comment by BobbyRyterski** [comment]
@jgrill, you might have the same problem that I saw. Check out #101 and see if you see SSH auth failures with debug level logging. I used `kitchen create -l debug`.

**Comment by jgrill** [comment]
Thanks @BobbyRyterski! #101 solved my problem with both 1.8.1 and 2.x!
